lexers/LexPython.cxx

```diff
@@ -714,9 +714,10 @@ void SCI_METHOD LexerPython::Lex(Sci_PositionU startPos, Sci_Position length, in
 							pos--;
 							ch = styler.SafeGetCharAt(pos, '\0');
 						}
-						if (pos < 0 || ch == '.') {
+						if (ch == '.') {
 							// Is this an attribute we could style? if it is, do as asked
-							bool isComment = false;
+							styler.Flush(); // need to be able to check to see if '.' is a comment or not, and need to flush the styler to check
+							bool isComment = (styler.StyleIndexAt(pos) == SCE_P_COMMENTLINE) || (styler.StyleIndexAt(pos) == SCE_P_COMMENTBLOCK);
```

Comment:
`AnyOf` could be used to avoid repeating the expression. Flushing the style buffer may have some performance impact and `BufferStyleAt` could be used like in cd45fdd.

Reply:
I had just noticed that `BufferStyleAt` in the other PR, and was finishing up switching to it as you replied here. :-)

I will see if I can switch to `AnyOf`